### PR TITLE
feat: added lerp interface

### PR DESCRIPTION
**Pull request by hexonaut**
Linear interpolation module interface.

### CODE REVIEW

File: src/dss/LerpAbstract.sol [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (brianmcmichael) [inline src/dss/LerpAbstract.sol:1]:
```suggestion
// SPDX-License-Identifier: AGPL-3.0-or-later
```

File: src/dss/LerpFactoryAbstract.sol [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (brianmcmichael) [inline src/dss/LerpFactoryAbstract.sol:1]:
```suggestion
// SPDX-License-Identifier: AGPL-3.0-or-later
```

File: src/dss/LerpAbstract.sol [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+// https://github.com/makerdao/dss-lerp/blob/master/src/Lerp.sol
+interface LerpAbstract {
+    function target() external view returns (address);
+    function what() external view returns (bytes32);
+    function start() external view returns (uint256);
+    function end() external view returns (uint256);
+    function duration() external view returns (uint256);
+    function done() external view returns (bool);
+    function startTime() external view returns (uint256);
+    function tick() external;
```
Comment (brianmcmichael) [inline src/dss/LerpAbstract.sol:13]:
```suggestion
    function tick() external;
    function ilk() external view returns (bytes32);
```

Comment (hexonaut) [reply src/dss/LerpAbstract.sol:13]: This only exists for the ilk lerp, but I suppose it's fine to include it.

**Review by brianmcmichael — APPROVED** [review]

**Pull request merged**